app/services/idv/analytics_events_enhancer.rb

```diff
@@ -1,61 +1,62 @@
-      idv_personal_key_visited
-      idv_phone_confirmation_form_submitted
-      idv_phone_confirmation_otp_rate_limit_attempts
-      idv_phone_confirmation_otp_rate_limit_locked_out
-      idv_phone_confirmation_otp_rate_limit_sends
-      idv_phone_confirmation_otp_resent
-      idv_phone_confirmation_otp_sent
-      idv_phone_confirmation_otp_submitted
-      idv_phone_confirmation_otp_visit
-      idv_phone_confirmation_vendor_submitted
-      idv_phone_error_visited
-      idv_phone_of_record_visited
-      idv_phone_otp_delivery_selection_visit
-      idv_phone_otp_delivery_selection_submitted
-      idv_proofing_resolution_result_missing
-      idv_enter_password_submitted
-      idv_enter_password_visited
-      idv_please_call_visited
-      idv_start_over
-    ].freeze
+    IGNORED_METHODS = %i[
+      idv_acuant_sdk_loaded
+      idv_back_image_added
+      idv_back_image_clicked
+      idv_barcode_warning_continue_clicked
+      idv_barcode_warning_retake_photos_clicked
+      idv_capture_troubleshooting_dismissed
+      idv_consent_checkbox_toggled
+      idv_exit_optional_questions
+      idv_front_image_added
+      idv_front_image_clicked
+      idv_image_capture_failed
+      idv_in_person_ready_to_verify_sp_link_clicked
+      idv_in_person_ready_to_verify_what_to_bring_link_clicked
+      idv_in_person_switch_back_submitted
+      idv_in_person_switch_back_visited
+      idv_link_sent_capture_doc_polling_complete
+      idv_link_sent_capture_doc_polling_started
+      idv_mobile_device_and_camera_check
+      idv_native_camera_forced
+      idv_verify_in_person_troubleshooting_option_clicked
+      idv_warning_action_triggered
+      idv_warning_shown
+    ].to_set.freeze
 
-    DECORATED_METHODS.each do |method_name|
-      define_method(method_name) do |**kwargs|
-        super(**kwargs, **common_analytics_attributes)
+    def self.included(_mod)
+      raise 'this mixin is intended to be prepended, not included'
+    end
+
+    def self.prepended(mod)
+      mod.instance_methods.each do |method_name|
+        if should_enhance_method?(method_name)
+          mod.define_method method_name do |**kwargs|
+            super(**kwargs, **common_analytics_attributes)
+          end
+        end
       end
     end
 
-    def self.included(_mod)
-      raise 'this mixin is intended to be prepended, not included'
+    def self.should_enhance_method?(method_name)
+      return false if IGNORED_METHODS.include?(method_name)
+
+      method_name.start_with?('idv_')
     end
 
     private
 
     def common_analytics_attributes
       {
         proofing_components: proofing_components,
-      }
+      }.compact
     end
 
     def proofing_components
       return if !user&.respond_to?(:proofing_component) || !user.proofing_component
-      ProofingComponentsLogging.new(user.proofing_component)
+      result = ProofingComponentsLogging.new(user.proofing_component).as_json
+      result.empty? ? nil : result
```

Comment:
can save a line or so with `#presence`
```suggestion
      ProofingComponentsLogging.new(user.proofing_component).as_json.presence
```